python/cuml/test/stemmer_tests/test_steps.py

```diff
@@ -0,0 +1,238 @@
+#
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cudf
+from cuml.preprocessing.text.stem.porter_stemmer import PorterStemmer
+
+
+def test_step1a():
+    word_str_ser = cudf.Series(
+        ["caresses", "ponies", "ties", "caress", "cats"]
+    )
+
+    st = PorterStemmer()
+    got = st._step1a(word_str_ser)
+
+    expect = ["caress", "poni", "tie", "caress", "cat"]
+    assert list(got.to_pandas().values) == expect
+
+
+def test_step1b():
+    word_str_ser_ls = [
```

Comment:
These tests seem to show a good diversity of inputs to stress the rules outlined in the docs for each of these steps. It would be nice if we could compare these directly against NLTK as well, or do you think that would be too heavy of a test dependency for cuml? 

It would be nice if there was a reasonable large list of words available that we could pass through both and verify the results match.

Reply:
So I have been testing against the `11,387` words that nltk provides for this to verify this.  See [gist](https://gist.github.com/VibhuJawa/d68f537547d245ad5d6e4ac314036bb1) . 

I will add that gist as a test too but it will mean adding `nltk` as a dependency for testing, Hoping that will be fine.

Reply:
Added test at: https://github.com/rapidsai/cuml/pull/2476/commits/e1593a465aa53dc19108dce9799f9f8400eb5c60 which should run once https://github.com/rapidsai/integration/pull/130 lands.